libcaf_core/caf/action.test.cpp

```diff
@@ -35,6 +38,25 @@ SCENARIO("actions wrap function calls") {
         check(uut.disposed());
       }
     }
+    WHEN("disposing an running action") {
+      auto called = false;
+      action* uut_ptr = nullptr;
+      auto uut = make_action([this, &called, &uut_ptr] {
+        called = true;
+        uut_ptr->dispose();
+        check_eq(uut_ptr->ptr()->current_state(),
+                 action::state::deferred_dispose);
+      });
+      uut_ptr = &uut;
+      THEN("it transitions to deferred_dispose") {
+        check(uut.scheduled());
+        uut.run();
+        check(called);
```

Comment:
Good idea writing the test like this to guarantee the action ending up the deferred dispose path. 👍

One nitpick: I'd keep the checks in the `THEN` block since we spell out the expected transition there but don't actually check it:

```suggestion
      });
      uut_ptr = &uut;
      THEN("it transitions to deferred_dispose") {
        check(uut.scheduled());
        uut.run();
        check(called);
        check_eq(uut_ptr->ptr()->current_state(),
                 action::state::deferred_dispose);
```

Reply:
I can't really refactor it like this since after finishing `run` the action would go to the `disposed` state. I need a check that is triggered during the `run` and asks for `deferred_dispose`.

Reply:
Yeah, good point. Didn't think of that. :)